### ISSUE TITLE
Not able to copy/dnd any FILE into VM, BUT text and picture in clipboard works fine

### ISSUE THREAD

**Issue by 424778940z** [vmware/open-vm-tools#605]
### Describe the bug

As title suggests, I'm not able to copy/drag files into the VM, but i can copy/drag files out, also any text/picture in clipboard works fine.

I'm using latest Xubuntu 22.04 LTS
I have verified vmtoolsd/open-vm-tools service is running.
Also open-vm-tools-desktop is installed

When im copy files in, following error shows up.
"GDBus.Error:org.gtk.GDBus.UnmappedGError.Quark._gnome_2drr_2derror_2dquark.Code17"

### Reproduction steps

```bash
1. install Xubuntu 22.04 LTS in VM
2. install open-vm-tools-desktop
3. copy file in
```

### Expected behavior

well, I expect the file being copied in....

**Comment by dsouders** [comment]
@424778940z

What host are you using?

**Comment by 424778940z** [comment]
> What host are you using?

Windows 10 LTSC 21H2

**Comment by dsouders** [comment]
Thanks!  I've filed an internal bug to track the issue.

**Comment by dsouders** [comment]
Oh, one more thing.  What version of workstation are you using?

**Comment by 424778940z** [comment]
> Oh, one more thing. What version of workstation are you using?

the latest 16.2.4

**Comment by dsouders** [comment]
Thanks!

**Comment by Normalmind1** [comment]
Is it solved?

**Comment by jonathanvmw** [comment]
@Normalmind1  - The bug is still being looked into.  Will post an update when more info is available.

**Comment by jonathanvmw** [comment]
@424778940z

The developer looking at this has a couple of follow-up questions:
- What version of tools is running?
- Is Xubuntu installed with X11 or Wayland?

Thanks.

**Comment by 424778940z** [comment]
> * What version of tools is running?
```
Package: open-vm-tools
Version: 2:12.3.5-3~ubuntu0.23.10.1
Package: open-vm-tools-desktop
Version: 2:12.3.5-3~ubuntu0.23.10.1
```

>Is Xubuntu installed with X11 or Wayland?
```
% env | grep -E -i 'x11|xorg|wayland'
CLUTTER_BACKEND=x11
XDG_SESSION_TYPE=x11
```

**Comment by 424778940z** [comment]
seems there is no `open-vmware-toolbox-cmd`, only `vmware-toolbox-cmd`
```
% vmware-toolbox-cmd -v 
12.3.5.46049 (build-22544099)
```

**Comment by karthigavmw** [comment]
@424778940z I tried dnd, copy/paste with WS 16.2.5 and the tools version same as yours - it works fine. And the only difference I see here is that you have mentioned that your GOS is Xubuntu 22.04. But in the package details you have mentioned - it says 
Package: open-vm-tools
Version: 2:12.3.5-3~**ubuntu0.23.10.1**
Package: open-vm-tools-desktop
Version: 2:12.3.5-3~**ubuntu0.23.10.1**


FYI - the package details I see in my VM is as below
Package :  open-vm-tools
Version :  2:12.3.5-3~**ubuntu0.22.04.1**
Package : open-vm-tools-desktop          
Version :  2:12.3.5-3~**ubuntu0.22.04.1**    

Can you please clarify on the version of GOS you are using here?

**Comment by 424778940z** [comment]
>Can you please clarify on the version of GOS you are using here?

I'm using Xubuntu, however, xubuntu is just an ubuntu that come with xfce desktop by default, they share the same package source/repository, I believe it is normal the package name only says ubuntu

**Comment by karthigavmw** [comment]
@424778940z I was asking it in order to reproduce the issue and check the logs as there aren't any attached in this bug. Also dnd, copy/paste works fine in our environment with Xubuntu 22.04.

**Comment by karthigavmw** [comment]
@424778940z I also tried it on WS 16.2.5 as well and it works fine for me. So can you please try it on the latest version of WS (17.5.1) or on 16.2.5? 

Also please make sure the OS installation and tools installation goes through properly. Because the error you mentioned in the bug is not internal.

**Comment by 424778940z** [comment]
Here are some update...
I'm on latest 17.5.1 now, and this issue still persists.

![image](https://github.com/vmware/open-vm-tools/assets/2379529/a654b5d0-8f4f-4ff1-9ce9-edc541763dc4)
a dump of clip board gives me this
```shell
adam@bzh-dev ~ % xclip -verbose -o --
Connected to X server.
Using UTF8_STRING.
Type is UTF8_STRING.
x-special/nautilus-clipboard
copy
file:///var/run/vmblock-fuse/blockdir/sakvBg/nR6a08e2.zip

```

**Comment by karthigavmw** [comment]
@424778940z I am assuming that you are using the same host version mentioned initially in the issue and the open-vm-tools is the latest. Please do correct me if I am wrong. 

Also if possible please collect and attach the log support bundle.

**Comment by 424778940z** [comment]
version dump

```shell
adam@bzh-dev ~ % sudo nala list --installed "open-vm-tools*" --verbose 
open-vm-tools 2:12.3.5-3~ubuntu0.23.10.1 [Ubuntu/mantic main]
├── is installed
└── Open VMware Tools for virtual machines hosted on VMware (CLI)

open-vm-tools-desktop 2:12.3.5-3~ubuntu0.23.10.1 [Ubuntu/mantic main]
├── is installed
└── Open VMware Tools for virtual machines hosted on VMware (GUI)
```
support bundle
[vm-2024-05-03.117372.tar.gz](https://github.com/vmware/open-vm-tools/files/15196463/vm-2024-05-03.117372.tar.gz)

**Comment by karthigavmw** [comment]
@424778940z Can you please let us know if you are observing this issue only when using nautilus or otherwise?

Dump of my clipboard is something like below and DnD/Copy works fine :

![image](https://github.com/vmware/open-vm-tools/assets/161797514/16a8d3dc-6786-4f67-93c5-cefe524e83e3)

**Comment by 424778940z** [comment]
>Can you please let us know if you are observing this issue only when using nautilus or otherwise?

welll, no, in face, i'm not even using nautilus at all, the default file manager for xface is thunar, and nautilus is not installed at all

**Comment by karthigavmw** [comment]
@424778940z Meanwhile, Can you please check if this workaround mentioned in https://askubuntu.com/questions/1276147/copied-filenames-prefixed-with-x-special-nautilus-clipboard-copy-file works for you ?

**Comment by 424778940z** [comment]
I think that workaround is for another issue, in my case, copy from vm to host works, but from host to vm don't, shortcut key seems not related

**Comment by karthigavmw** [comment]
> I think that workaround is for another issue, in my case, copy from vm to host works, but from host to vm don't, shortcut key seems not related

yeah I got that...wondering about the difference in our clipboard dumps (x-special/nautilus-clipboard).

**Comment by karthigavmw** [comment]
@424778940z We were not able to reproduce the issue in our environments with same HOS and GOS as yours. Tried both Dnd and Copy/Paste and it works fine with no issues.

1. Copy a folder from HOS
2. Do a paste on the GOS which is successful.
3. Call xclip -verbose -o -- . The clipboard dump doesn't contain "x-special/nautilus-clipboard" in this case. The dump observed was as mentioned above comments of mine (https://github.com/vmware/open-vm-tools/issues/605#issuecomment-2097803486)

However, after multiple trials, we observed the issue reproducing only for copy/paste of folder (file copy still works fine though) by following the below steps :
1. Copy a folder from HOS
2. Call the xclip -verbose -o --
4. Observe that the dump contains "x-special/nautilus-clipboard"
5. Then do a paste on the GOS. Observed the same error you mentioned.
  ----------> Upon debugging this further we found that the read system call is taking more time (around 3 or 5 min) to complete when reading from the staging directory in GOS and thats the reason paste is failing in our case. As its a system call we couldn't do anything else in this case as calling read is very much necessary here.

**Comment by karthigavmw** [comment]
@424778940z Can you please enable debug logs in your environment and share those to check on what is causing issue in your setup. To enable debug logs for open-vm-tools, please refer to https://knowledge.broadcom.com/external/article/313512/enabling-debug-logging-for-vmware-tools.html
